pkg/azure/metrics/metrics.go

```diff
@@ -0,0 +1,291 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
One more thing I'm wondering about - is making this Azure specific the correct thing to do here?  Presumably other platforms have their own rate limiting.  I wonder if we should just have a generic rate limit metric for platform API requests.